packages/checkout-storefront/src/lib/utils/locale.ts

```diff
@@ -18,3 +19,12 @@ export const getParsedLocaleData = (
     languageCode: languageCode as LanguageCodeEnum,
   };
 };
+
+export const createGetCountryNames = () => {
+  const countryNames = new Intl.DisplayNames(
+    [getParsedLocaleData(getQueryParams().locale).languageCode],
+    { type: "region" }
+  );
+
+  return (countryCode: CountryCode): string => countryNames.of(countryCode) as string;
```

Comment:
Maybe instead of `as string` we should fall back to returning `countryCode`?
```suggestion
  return (countryCode: CountryCode): string => countryNames.of(countryCode) || countryCode;
```

Reply:
Sure, why not 👍